sktime/forecasting/base/adapters/_pmdarima.py

```diff
@@ -89,17 +111,43 @@ def _predict_in_sample(
 
         fh_abs = fh.to_absolute(self.cutoff)
         fh_idx = fh.to_indexer(self.cutoff, from_cutoff=False)
+
+        # bacause indexer will return 0 for first train point we substract
+        # the differencing order to make 0 the index of the first predictable point
+        if return_pred_na:
+            fh_nan = fh_idx[fh_idx < self.order[1]]
+            fh_idx = fh_idx[fh_idx >= self.order[1]] - self.order[1]
+
         if return_pred_int:
             # unpack and format results
             y_pred, pred_int = result
-            y_pred = pd.Series(y_pred[fh_idx], index=fh_abs)
-            pred_int = pd.DataFrame(
-                pred_int[fh_idx, :], index=fh_abs, columns=["lower", "upper"]
-            )
+            if return_pred_na:
```

Comment:
it would be easier to read to keep the old code, and then have one block `if return_pred_na` which uses `np.concatenate` to pre-pend `y_pred = np.concatenate(stuff, y_pred)`

Reply:
Yes sounds cleaner and more understandable. I will do that.

Reply:
this isn't done yet, is it?

Reply:
I hope this is what you had in mind ^^

Reply:
no, not quite - I was asking you to reduce nesting layer and repetition of `if`-s, or this will be hard to read and debug/extend.

In pseudo-code:

1. compute `y_pred`
2. if `return_pred_na`, adapt the `fh_` variables
3. if `return_pred_int`, compute prediction intervals (without NAs)
4. if `return_pred_na`, prepend NAs to `y_pred` and `pred_int` (if exists)
5. return `y_pred` and possibly `pred_int` (if `return_pred_int`)

Maybe it's also possible to reduce the `if return_pred_na` blocks to a single one?

Reply:
for instance, initializing `pd.Series` and `pd.DataFrame` with all-NA at the start, and only filling in the "valid" predictions? That would avoid index magic and prepending etc.

Reply:
Actually it is possible since the NA series are made with length fh_na which can be zero, I can always prepend it without verifying the condition.

Reply:
great!
looking forward to it.

Reply:
I guess It looks cleaner now, thou Ihad to rearrange the order of some stuff for less code and parameters.

Reply:
Hm yes - though the tests fail now...

The tests that fail check whether the indices that are being predicted are indeed those that predictions were asked for. Probably some minor bug in there?

Reply:
ooh it's not actually that, it's just that I did my tests only with ARIMA while AutoARIMA hides its order under another attribute layer (`self._forecaster.model_.order`). I'll fix it right away.